src/main/java/com/basho/riak/client/core/RiakCluster.java

```diff
@@ -636,6 +636,21 @@ public static Builder builder(RiakNode node)
         }
 
     /**
+     * Cleans up any Thread-Local variables after shutdown.
+     * This operation is useful when you are in a container environment, and you
+     * do not want to leave the thread local variables in the threads you do not manage.
+     * Call this method when your application is being unloaded from the container, <b>after</b>
+     * all {@link RiakNode}, {@link RiakCluster}, and {@link com.basho.riak.client.api.RiakClient}
+     * objects are in the shutdown state.
+     */
+    public synchronized void cleanup()
```

Comment:
Nah, b/c:
1. If "the user" instantiated 2 RiakCluster or RiakClient objects for some reason, it could clear all the Netty threadlocals if one is still running. 
2. It would turn `shutdown()` into a partially synchronous method, as we'd have to wait for the shutdown future to finish, then call cleanup. 
3. This is only for environments that swap out classloaders.